daemon/api_model.go

```diff
@@ -69,3 +89,90 @@ func postModel(c *Command, r *http.Request, _ *auth.UserState) Response {
 	return AsyncResponse(nil, &Meta{Change: chg.ID()})
 
 }
+
+// helper for parsing url options into formatting option vars
+func parseHeadersFormatOptionsFromURL(url *url.URL) (
+	jsonResult bool,
+	headersOnly bool,
+	headers map[string]string,
+	err error,
+) {
+	headers = map[string]string{}
+	q := url.Query()
+	for k := range q {
+		if k == "json" {
+			switch q.Get(k) {
+			case "false":
+				jsonResult = false
+			case "headers":
+				headersOnly = true
+				fallthrough
+			case "true":
+				jsonResult = true
+			default:
+				err = errors.New(`"json" query parameter when used must be set to "true" or "headers"`)
+				return jsonResult, headersOnly, headers, err
+			}
+			continue
+		}
+		headers[k] = q.Get(k)
```

Comment:
what use are these `headers`? you're not using them.

Reply:
They are used in the /v2/assertions endpoint which I have refactored to use this helper